react/features/facial-recognition/facialExpressionsWorker.js

```diff
@@ -1,120 +1,164 @@
-// @flow
 import './faceApiPatch';
+
+import { setWasmPaths } from '@tensorflow/tfjs-backend-wasm';
 import * as faceapi from '@vladmandic/face-api';
 
-import {
-    CLEAR_TIMEOUT,
-    CPU_TIME_INTERVAL,
-    FACIAL_EXPRESSION_MESSAGE,
-    INIT_WORKER,
-    SET_TIMEOUT,
-    INTERVAL_MESSAGE,
-    WEBGL_TIME_INTERVAL
-} from './constants';
+import { DETECTION_TYPES, DETECT_FACE, FACE_BOX_MESSAGE, FACIAL_EXPRESSION_MESSAGE, INIT_WORKER } from './constants';
 
 /**
- * A flag that indicates whether the tensorflow models were loaded or not.
+ * Detection types to be applied.
  */
-let modelsLoaded = false;
+let faceDetectionTypes = [];
+
+/**
+ * Indicates whether an init error occured.
+ */
+let initError = false;
 
 /**
- * The url where the models for the facial detection of expressions are located.
+ * A flag that indicates whether the models are loaded or not.
  */
-let modelsURL;
+let modelsLoaded = false;
 
 /**
  * A flag that indicates whether the tensorflow backend is set or not.
  */
 let backendSet = false;
 
 /**
- * A timer variable for set interval.
+ * Flag for indicating whether an init operation (setting tf backend, loading models) is in progress.
  */
-let timer;
+let initInProgress = false;
 
 /**
- * The duration of the set timeout.
+ * Flag for indicating whether a face detection flow is in progress or not.
  */
-let timeoutDuration = -1;
+let detectionInProgress = false;
 
 /**
- * A patch for having window object in the worker.
+ * Contains the last valid face bounding box (passes threshold validation) which was sent to the main process.
  */
-const window = {
-    screen: {
-        width: 1280,
-        height: 720
+let lastValidFaceBox;
+
+const detectFaceBox = async ({ imageTensor, threshold }) => {
+    const detections = await faceapi.detectAllFaces(imageTensor, new faceapi.TinyFaceDetectorOptions());
+
+    if (!detections.length) {
+        return;
+    }
+
+    const faceBox = {
```

Comment:
IF multiple faces are detected we are taking a box that covers them all right?

Reply:
yes